File: temp/test-scripts/test_ws.py
import websocket
import ssl
import json

token = 'eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJhZG1pbiIsImxvZ2luX3VzZXJfa2V5IjoiYTQ0NDkzZjQtODA1My00N2Y5LTgwMzQtNDY2MWJiZGM3N2QwIn0.1pV34Pk9xzjONWHnG_7UDIJz6U-SUDvEpp3WvNy0C2owVEzraZS8X57y1QoPA9q5eI2m0hnbJX_K8jyf_-jvCw'
conversation_id = 'test-123'
ws_url = f'ws://localhost:8080/ws/chat/{conversation_id}?token={token}'

print(f'Connecting to: {ws_url}')

try:
    ws = websocket.create_connection(
        ws_url,
        timeout=10,
        header=[
            f"Cookie: Admin-Token={token}"
        ]
    )
    print('WebSocket connected successfully!')
    print(f'Protocol: {ws.getstatus()}')
    ws.close()
except Exception as e:
    print(f'Error: {type(e).__name__}: {e}')

File: temp/test-scripts/test_ws.js
// WebSocket connection test script
const token = 'eyJhbGciOiJIUzUxMiJ9.eyJzdWIiOiJhZG1pbiIsImxvZ2luX3VzZXJfa2V5IjoiYTQ0NDkzZjQtODA1My00N2Y5LTgwMzQtNDY2MWJiZGM3N2QwIn0.1pV34Pk9xzjONWHnG_7UDIJz6U-SUDvEpp3WvNy0C2owVEzraZS8X57y1QoPA9q5eI2m0hnbJX_K8jyf_-jvCw';
const conversationId = 'test-123';
const wsUrl = `ws://localhost:8080/ws/chat/${conversationId}?token=${token}`;

console.log('Connecting to:', wsUrl);

const ws = new WebSocket(wsUrl);

ws.onopen = () => {
  console.log('WebSocket connected successfully!');
  console.log('Protocol:', ws.protocol);
  console.log('ReadyState:', ws.readyState);
  ws.close();
  process.exit(0);
};

ws.onerror = (error) => {
  console.log('WebSocket error:', error.message || error);
  process.exit(1);
};

ws.onclose = (event) => {
  console.log('WebSocket closed:', event.code, event.reason);
  if (event.code !== 1000) {
    process.exit(1);
  }
};

// Timeout after 10 seconds
setTimeout(() => {
  console.log('Connection timeout');
  ws.close();
  process.exit(1);
}, 10000);
